nodejs/eks/nodes/nodegroup.ts

```diff
@@ -572,10 +571,10 @@ export function resolveInstanceProfileName(
     } else if (c.nodeGroupOptions.instanceProfileName) {
         return pulumi.output(c.nodeGroupOptions.instanceProfileName!);
     } else {
-        throw new pulumi.ResourceError(
-            `an instanceProfile or instanceProfileName is required`,
-            parent,
-        );
+        throw new pulumi.InputPropertyError({
```

Comment:
In these cases where the error could be attributed to multiple properties should we add two property errors or should it stay as a resource error?

Reply:
I feel like a single input property error is enough here.
The benefit of the input property error is that a user will be presented with all errors and not only the first validation that failed.

**For example compare those two errors for the same program:**

Old:
<img width="1506" alt="Screenshot 2025-01-23 at 11 01 38" src="https://github.com/user-attachments/assets/260a3467-100d-4ba7-9871-6d852ac8e2a7" />
New:
<img width="1473" alt="Screenshot 2025-01-23 at 10 59 25" src="https://github.com/user-attachments/assets/17db2da0-6fd6-40a3-bc0a-8661bd8ee26a" />

IMO we shouldn't create multiple errors for the same issue when there's multiple properties involved. That's just gonna increase the cognitive complexity of understanding the issue.

Reply:
ah I see. I wasn't sure how the property errors would be shown, but since they are just listed out under the resource then that makes total sense.

It looks like that value `{{urn:pulumi:..}` is something that the engine adds? I'm assuming it only works with primitive types or something?

Reply:
That's how dependency resources are visualized here. I guess the URN is the only sensible value that can be shown here 🤔

Reply:
Just curious, does it resolve `Output` values correctly? Wondering if we ever need to be careful about what kind of values we can include here.

Reply:
You mean the message the engine adds, right? No, this one shows either that it's an output (e.g. during preview) or the resolved value if it's already resolved.